launcher/__main__.py

```diff
@@ -3,31 +3,49 @@
 import os
 import sys
 import argparse
-import PyQt5
+import importlib
 
 EXIT_SUCCESS = 0
 EXIT_FAILURE = 1
 
 
 def cli():
-    # External Dependencies
+    # Check environment dependencies
+    missing = [
+        dep not in os.environ
+        for dep in (
+            "AVALON_CONFIG",
+            "AVALON_PROJECTS"
+        )
+    ]
+
+    if missing:
+        sys.stderr.write(
+            "Incomplete environment, missing variables:\n%s"
+            % "\n".join("- %s" % var for var in missing)
+        )
+
+        return EXIT_FAILURE
+
+    # Check modules dependencies
     missing = list()
-    dependencies = (
-        "PYBLISH_BASE",
-        "PYBLISH_QML",
-        "AVALON_CONFIG",
-        "AVALON_PROJECTS",
-        "AVALON_CORE",
-    )
+    dependencies = {
+        "PyQt5": None,
+        "avalon": None
+    }
 
     for dependency in dependencies:
-        if dependency not in os.environ:
+        try:
+            dependencies[dependency] = importlib.import_module(dependency)
```

Comment:
Removed `__import__` as it's use is generally discouraged. Also removed `finally` as it would have thrown an error since `lib` isn't declared if the try/except fails, but finally would have run either way.

Reply:
Nice! Thanks :)